iFixit/react-commerce
# Product list swr cache

- Adds a [dev dependency](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript-validation-schema) ([GitHub](https://github.com/Code-Hex/graphql-codegen-typescript-validation-schema/tree/main)) to generate zod schemas for all inputs in Strapi's graphql schema.
- Adds a swr-cache endpoint to cache the product list model, with a 1 hour TTL and a 1 day SWR.

Closes https://github.com/iFixit/react-commerce/issues/1612

## QA

- Smoke test the product list pages to make sure they match main and don't produce any new errors in the logs or Sentry.
- While visiting the product list pages, open the Vercel logs and confirm that we get cache misses, stales, and hits when appropriate based on the timings listed above.
